app/controllers/thesis_controller.rb

```diff
@@ -44,6 +44,19 @@ def edit
     @thesis.association(:advisors).add_to_target(Advisor.new) if @thesis.advisors.count.zero?
   end
 
+  def publish_to_dspace
+    # This will be where the publication job gets invoked.
```

Comment:
This made me realize the publishing job currently publishes one thesis, but the expectations here are that we are publishing a specific set. While we could loop over the set here and spin up 1000 jobs, we might also considering refactoring the publishing job to do that work as I'm worried spinning up 1000 jobs might take longer than the heroku timeout and itself need to be backgrounded. <sigh>